api/storage_service.cc

```diff
@@ -1552,6 +1554,26 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         co_return json_void();
     });
 
+    ss::repair_tablet.set(r, [&ctx, &ss] (std::unique_ptr<http::request> req) -> future<json_return_type> {
+        auto tokens_param = split(req->get_query_param("tokens"), ",");
+        utils::chunked_vector<dht::token> tokens;
+        bool all_tokens = tokens_param.size() == 1 && tokens_param.front() == "all";
+        if (!all_tokens) {
+            tokens.reserve(tokens_param.size());
+            for (auto& t : tokens_param) {
+                auto token = dht::token::from_int64(validate_int(t));
+                tokens.push_back(token);
+            }
+        }
+        auto ks = req->get_query_param("ks");
+        auto table = req->get_query_param("table");
+        validate_table(ctx, ks, table);
+        auto table_id = ctx.db.local().find_column_family(ks, table).schema()->id();
+
+        auto res = co_await ss.local().add_repair_tablet_request(table_id, std::move(tokens), all_tokens);
```

Comment:
Like I mentioned above, I want to avoid reuse empty of the tokens to a special meaning. If the user requested all tokens, make the request explicit. 

Btw, if we want to make empty to present all, we could make the tokens vector empty to avoid the std::optional.

Reply:
This suggestion is independent of the API change. It's only about the implementation.

There is no need for extra parameter all_tokens, because it's not independent of "tokens". We either have a list of tokens, or "all_tokens = true". If all_tokens is true, the list is ignored. It's cleaner to express this. So it's `variant<std::vector<token>, all_tokens::yes>`, which can be expressed simply `optional<std::vector<token>>` where nullopt means all_tokens == true. 

> Btw, if we want to make empty to present all, we could make the tokens vector empty to avoid the std::optional.

Empty vector should mean no tokens, otherwise it's inconsistent with the algebra. For example, if tokens come from `intersection(tokens_0, tokens_1)`, you would have to special case for empty result, where the intent is to repair nothing rather than everything.